source/DataFile.cpp

```diff
@@ -50,8 +50,8 @@ void DataFile::Load(const string &path)
 		data.push_back('\n');
 
 	// Note what file this node is in, so it will show up in error traces.
-	root.tokens.push_back("file");
-	root.tokens.push_back(path);
+	root.AddToken("file");
+	root.AddToken(path);
```

Comment:
This looks like a good change, that is completely unrelated to this PR. I suggest to keep this for a separate PR (since this PR is already quite complex with all the changes that are relevant).

Reply:
I removed all the DataFile changes.